igor-web/src/main/groovy/com/netflix/spinnaker/igor/pubsub/PubsubSubscriber.groovy

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.pubsub
+
+import com.netflix.spinnaker.igor.model.PubsubType
+
+import javax.naming.OperationNotSupportedException
+
+abstract class PubsubSubscriber {
+  static PubsubType pubsubType() {
```

Comment:
Is this to distinguish between `PubsubSubscriber` types? If so, why not check the `.class`?